### PR TITLE
Shallow support for Cmdr conditional arguments

### PR DESCRIPTION
**Pull request by Kiwi-i**
Allows for the conditional (function) arguments in evaera/Cmdr to be properly registered by the client and thus work as expected, also allows `cmds` to still run without throwing errors, and at least displays some hint that there is additional arguments.

Proposed fix for #17

### CODE REVIEW

File: src/Client/ClientRegistry.lua [hunk 1]
```diff
@@ -56,6 +56,16 @@ function ClientRegistry:LoadCommand(CommandData: Types.NexusAdminCommandData): (
     elseif ExistingCommand then
         CmdrCommandData.ClientRun = ExistingCommand.ClientRun
     end
+
+    -- Force arguments lost during invocation to be included on the client. Extremely shallow.
+	if ExistingCommand then
+		for index,arg in ipairs(ExistingCommand.Args) do
+			if typeof(CmdrCommandData.Args[index]) ~= typeof(arg) then
+				table.insert(CmdrCommandData.Args,index,arg)
```
Comment (TheNexusAvenger) [inline src/Client/ClientRegistry.lua]: Formatting issues:
- PascalCase is used in the project instead of camelCase.
- `ipairs` isn't used in any other part of the project. While it is more performant, it isn't enough for this application ro demand it.
- `i` is used for indexing by number in other parts of the project.
- Full names are preferred, such as `Argument` over `arg`.
- While this file isn't consistent, new changes are meant to have a space after a comma.

Requested lines:
```lua
for i, Argument in ExistingCommand.Args do
    if typeof(CmdrCommandData.Args[i]) ~= typeof(Argument ) then
        table.insert(CmdrCommandData.Args, i, Argument )
```

File: src/IncludedCommands/Administrative/cmds.lua [hunk 2]
```diff
@@ -44,11 +44,15 @@ return {
                         for _,Prefix in Prefixes do
                             local CommandString = Prefix..CmdrCommand.Name.." "
                             for _,Argument in CmdrCommand.Args do
-                                if Argument.Optional == true then
-                                    CommandString = CommandString.."("..Argument.Name..") "
-                                else
-                                    CommandString = CommandString..Argument.Name.." "
-                                end
+                                if typeof(Argument)=='function' then
```
Comment (TheNexusAvenger) [inline src/IncludedCommands/Administrative/cmds.lua]: Formatting issues:
- Spaces are used surrounding `==` in every other case of the project.
- Double quotes are used for all other strings.

Requested line:
```lua
if typeof(Argument) == "function" then
```

**Review by TheNexusAvenger — CHANGES_REQUESTED** [review]
While I haven't tested it, the logic seems fine. The formatting isn't right. The main thing is that the project uses 4 spaces for indents while the pull request uses tabs. Other formatting changes are requested below.

**Comment by Kiwi-i** [comment]
My bad, was lazy and used the github web editor and forgot to change the tab settings...

**Comment by TheNexusAvenger** [comment]
Formatting looks good. I am going to be a bit delayed to test the functionality, but it should be before the end of the week.

**Comment by Kiwi-i** [comment]
No worries, so far I've had no issues with it, but certainly have not tested any edge cases.

**Review by TheNexusAvenger — APPROVED** [review]

**Comment by TheNexusAvenger** [comment]
Nexus Admin V.2.7.6 was **rolled back** due to this change. If a command was overwritten on the server and then the original command is registered on the client, then the extra arguments from that command will mutate the overwritten command.
